polyfill/lib/ecmascript.mjs

```diff
@@ -793,15 +918,15 @@ export const ES = ObjectAssign({}, ES2020, {
     if (validUnits.indexOf(unit1) > validUnits.indexOf(unit2)) return unit2;
     return unit1;
   },
-  ToPartialRecord: (bag, fields) => {
+  ToPartialRecord: (bag, fields, cast = ES.ToInteger) => {
```

Comment:
I'm not sure the spec language has first-class abstract operations like this, but, maybe consider instead splitting `BUILTIN_FIELDS` into `BUILTIN_DATETIME_FIELDS` and `DURATION_FIELDS`, and using `ToNumber` on duration fields?